### PR TITLE
Implement servomotors handling for actuator board

### PR DESCRIPTION
**Pull request by antoinealb**
This pull request implements control of the servomotor output that are on the actuator board. They can be controlled through UAVCAN and support having a smooth acceleration.

Tested on hardware (with a frequency meter and voltmeter as I don't have any RC servos).

Note that this PR depends on #267 as it uses the new DEBUG_N calls.

### CODE REVIEW

File: actuator-firmware/src/pwm.h [hunk 1]
```diff
@@ -8,17 +8,32 @@ extern "C" {
...
+                        uint32_t timer_period);
 
-/* Set given pulse width in clock ticks to the corresponding PWM channel */
-void pwm_set_pulsewidth(enum pwm_channel channel, uint32_t pulsewidth);
+/** Sets the pulse width for that particular timer and channel.
+ *
+ * Note that the channel number is 0-indexed here, while in the datasheet it is
+ * 1-indexed.
+ */
+void pwm_set_pulsewidth(stm32_tim_t* tim, uint8_t channel, uint32_t width);
```
Comment (SyrianSpock) [inline actuator-firmware/src/pwm.h]: Maybe we want also an API with on/off and configurable pwm for those

Comment (antoinealb) [reply actuator-firmware/src/pwm.h]: What would you want to configure ?

Comment (SyrianSpock) [reply actuator-firmware/src/pwm.h]: on/off pulsewidth

Comment (antoinealb) [reply actuator-firmware/src/pwm.h]: Well the on pulse width is specified from the duty cycle, so once per channel (hence this function). On the other hand, the cycle duration can only be specified on a per-timer basis (i.e. for all channels at once), and this is why it is provided as a parameter to `pwm_setup_channels`.

File: actuator-firmware/src/uavcan/Command_handler.cpp [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+#include <cvra/actuator/Command.hpp>
+
+#include "uavcan/build_config.hpp"
+
+int Command_handler_start(uavcan::INode& node)
+{
+    int ret;
+    static uavcan::Subscriber<cvra::actuator::Command> sub(node);
+    ret = sub.start(
```
Comment (nuft) [inline actuator-firmware/src/uavcan/Command_handler.cpp]: It's good practice to initialize local variables at declaration. Or in this case, return directly.
And always use const when possible.

File: actuator-firmware/src/servo.c [hunk 3]
```diff
@@ -100,6 +95,9 @@ THD_FUNCTION(servo_thd, arg)
 
 void servo_start(void)
 {
-    static THD_WORKING_AREA(servo_thd_wa, 256);
+    static THD_WORKING_AREA(servo_thd_wa, 512);
+
+    servo_init();
```
Comment (nuft) [inline actuator-firmware/src/servo.c]: `servo_init()` -> `timer_init()`

File: actuator-firmware/src/servo.c [hunk 4]
```diff
@@ -100,6 +95,9 @@ THD_FUNCTION(servo_thd, arg)
 
 void servo_start(void)
 {
-    static THD_WORKING_AREA(servo_thd_wa, 256);
+    static THD_WORKING_AREA(servo_thd_wa, 512);
```
Comment (nuft) [inline actuator-firmware/src/servo.c]: We should monitor actual stack usage and not just randomly double.
This is ok for development but we might run into memory shortage on this constrained MCU.

File: actuator-firmware/src/uavcan/Command_handler.cpp [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+                return;
+            }
+            /* TODO: Implement a watchdog to stop the movements if no setpoints come from master */
+            /* TODO: Handle commands other than servomotors. */
+
+            UAVCAN_ASSERT(SERVO_COUNT == msg.servo_trajectories.size());
+
+            for (int i = 0; i < SERVO_COUNT; i++) {
+                servo_set(i, msg.servo_trajectories[i].position, msg.servo_trajectories[i].velocity, msg.servo_trajectories[i].acceleration);
+                DEBUG_N(10, "received setpoint %03d for servo %d", static_cast<int>(msg.servo_trajectories[i].position * 1000), i);
```
Comment (nuft) [inline actuator-firmware/src/uavcan/Command_handler.cpp]: DEBUG_N -> DEBUG_EVERY_N

File: actuator-firmware/src/uavcan/Command_handler.cpp (hunk 2 above)
Comment (antoinealb) [reply actuator-firmware/src/uavcan/Command_handler.cpp]: done.

File: actuator-firmware/src/servo.c (hunk 3 above)
Comment (antoinealb) [reply actuator-firmware/src/servo.c]: servo_init does more than timer_init, and is a separate function.

File: actuator-firmware/src/servo.c (hunk 4 above)
Comment (antoinealb) [reply actuator-firmware/src/servo.c]: +1, not even sure that stack size is an issue here.

File: actuator-firmware/src/uavcan/Command_handler.cpp (hunk 5 above)
Comment (antoinealb) [reply actuator-firmware/src/uavcan/Command_handler.cpp]: done

**Pull request merged**